Esd   add/modify/delete accounts

Hyperoid29 opened the pull request:
Added both templates in which uses a CRUD functionality to add/modify and delete both clubs and user accounts. Created multiple functions within 'views.py', that operates either adding, modifying or deleting the account or club, depending on what the account manager has clicked in.

Hyperoid29 committed:
Created two CRUD functionality templates

Created two templates in which uses a CRUD functionality to add/modify and delete either the club accounts or the account management.

Hyperoid29 committed:
Created additional URL's (add/delete/modify). 

Created multiple URL paths for the CRUD functionality. These functionalities consists of adding, updating and deleting user or club.

Hyperoid29 committed:
Added views for each CRUD functionality (add, modify and delete).

The changes that I have made is the CRUD functionalities such as adding, modifying and deleting accounts or clubs. Each of these functions linked towards both 'manage_accounts' and 'manage_club_accounts'.

Hyperoid29 committed:
Replaced 'AmendAccounts.html' template

I replaced the amend accounts template with the two new templates I created. These two templates are 'ManageAccounts' and 'Manage Club Accounts'.

ChristianECDawson commented on Cinema-System-ESD-Component B/UWEFlix_django/uweflix/views.py:
- [ ] Minor : Consider using Django's built-in ListView, CreateView, UpdateView, and DeleteView to handle basic CRUD operations. These views can save you time and effort and are generally more secure and maintainable than custom views.

- [x] Major : It would be good to include some error handling to handle cases where the user does not exist or there is an error while adding/updating/deleting a user or a club. For example, you could use try-except blocks to catch exceptions and return appropriate error messages.

ChristianECDawson reviewed (commented):
The code looks good!
A couple of thoughts on how to reduce boilerplate code with Django's built-in ListView, CreateView, etc. Also, should probably add some error handling around functions that access the database in case of nonexistent users.

Hyperoid29 committed:
Updated URLS.py

Added extra import 'from .views import *'

alamerton reviewed (approved):
Looks good 👍

Hyperoid29 committed:
Updated AccessClubForm in forms.py

Updated AccessClubForm to update chosen clubs payment details. The club list will be automatically updated after the club has been added.

Hyperoid29 committed:
Updated views.py

Added extra functionality for updating payment details as well as try catch exception handling.

Hyperoid29 committed:
Updated manage_club_account.html

Replaced  name="club" to name="name"  in line 81

Hyperoid29 committed:
Update views.py 

Added 'import re' for email handling.

Hyperoid29 committed:
Updated set_payment.html

After updating payment, a message will pop-up regarding successful change.

Hyperoid29 committed:
Updated am_home.html

Changed 'account_modify' to amend user account and clubs.
